internal/utils/gatewayclass/get.go

```diff
@@ -30,6 +35,10 @@ func Get(ctx context.Context, cl client.Client, gatewayClassName string) (*Decor
 			vars.ControllerName(),
 		))
 	}
+	acceptedCondition, found := k8sutils.GetCondition(consts.ConditionType(gatewayv1.GatewayClassConditionStatusAccepted), gwc)
+	if acceptedCondition.Status != metav1.ConditionTrue || !found {
```

Comment:
```suggestion
	if !found || acceptedCondition.Status != metav1.ConditionTrue {
```
I suggest this order, because as I understand it, when `!found` is `true` value of `acceptedCondition` should be discarded